From cbb518d008e291597519e3aefd1671011e1ba46a Mon Sep 17 00:00:00 2001
From: "Bhadouria, Aman" <aman.bhadouria@intel.com>
Date: Fri, 4 Oct 2024 08:06:13 +0000
Subject: [PATCH] Fix for Android 15 Ip Provisioning Failure

In Android 15, Wifi is not working due to ip provisioning
failure. This is because of DHCP Accept packet being filtered
out by DhcpFilter from jni level.

By checking the DHCP handling logic from Android 14, which does
not include the problematic macros, we found that one of the
macro had wrong jump statment. Now, DHCP accept packets are not
filtered and can reach the router.

Tests Done:
1. Boot android image for both caas and base_aaos
   with this change
2. Go to settings and Connect to Wifi from router
   and from Mobile Hotspot
3. Wifi Connects Succesfully

Tracked-On: OAM-129211
Signed-off-by: Bhadouria, Aman <aman.bhadouria@intel.com>
Co-authored-by: Panda, Bharat B <bharat.b.panda@intel.com>
Co-authored-by: Gowtham Anandha Babu <gowtham.anandha.babu@intel.com>
---
 jni/network_stack_utils_jni.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/jni/network_stack_utils_jni.cpp b/jni/network_stack_utils_jni.cpp
index 6f47d7ef..7215e8ff 100644
--- a/jni/network_stack_utils_jni.cpp
+++ b/jni/network_stack_utils_jni.cpp
@@ -97,7 +97,7 @@ static void network_stack_utils_attachDhcpFilter(JNIEnv *env, jclass clazz, jobj
     static sock_filter filter_code[] = {
         // Check the protocol is UDP.
         BPF_LOAD_IPV4_U8(protocol),
-        BPF2_REJECT_IF_NOT_EQUAL(IPPROTO_UDP),
+        BPF2_ACCEPT_IF_EQUAL(IPPROTO_UDP),
 
         // Check this is not a fragment.
         BPF_LOAD_IPV4_BE16(frag_off),
-- 
2.34.1

